src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/TypeMaker.java

```diff
@@ -234,6 +234,14 @@ public Type visitArray(Type.Array t, Void aVoid) {
             return Type.pointer(t.elementType());
         }
 
+        @Override
+        public Type visitDelegated(Type.Delegated t, Void aVoid) {
+            if (t.kind() == Delegated.Kind.TYPEDEF && t.type() instanceof Type.Array) {
+                return visitArray((Type.Array)t.type(), aVoid);
```

Comment:
shouldn't we just recurse on the type() field? what if it's a typedef of a typedef of a typedef....

Reply:
type() seems to return canonical type. I can another test for typedef of typedef.